### PR TITLE
Release version 4.2.0

### PR DESCRIPTION
**Pull request by guialbuk**
### Features

* Add support for `optional: false` in `belongs_to` associations. ([#1237])
* Add support for associations with `inverse_of: false` and non-standard foreign key ([#1106])

### Bug fixes
* Fix typos in documentation of `allow_values` ([#1241])
* Fix appraisal command in CONTRIBUTING.md file ([#1253])

### Improvements
* Remove minitest reporters dependency ([#1251])
* Development dependency updates

[#1106]: https://github.com/thoughtbot/shoulda-matchers/pull/1106
[#1237]: https://github.com/thoughtbot/shoulda-matchers/pull/1237
[#1241]: https://github.com/thoughtbot/shoulda-matchers/pull/1241
[#1253]: https://github.com/thoughtbot/shoulda-matchers/pull/1253

### CODE REVIEW

File: .tool-versions [hunk 1]
```diff
@@ -0,0 +1 @@
+ruby 2.6.5
```
Comment (mcmire) [inline .tool-versions]: Ooh what is this??

Comment (guialbuk) [reply .tool-versions]: This is for `asdf` and other tools that manage versions for different languages

**Comment by guialbuk** [comment]
> This looks good. Make sure to re-run `appraisal install` locally as the version shows up in all of the generated gemfiles as well. Other than that, feel free to release!

Hi @mcmire . I was running `appraisal install` but there are no `shoulda-matcher` versions in the generated Gemfiles, as unit tests directly require the gem code.

For acceptance tests, the gem is directly loaded from the filesystem:

https://github.com/thoughtbot/shoulda-matchers/blob/e0a0200fe47157c161fb206043540804bdad664e/spec/support/acceptance/adds_shoulda_matchers_to_project.rb#L32

So I think we don't need to update any generated gemfiles. Or maybe I'm missing something?

File: .tool-versions (hunk 1 above)
Comment (O330oei) [reply .tool-versions]: Mange Is to manage other languages

Comment (guialbuk) [reply .tool-versions]: I removed `.tool-version` as it's kind of redundant. Multiple language manages can fallback to ` .ruby-version `.

Comment (mcmire) [reply .tool-versions]: Ok! No worries. I was just wondering! Regardless `asdf` seems like a pretty cool tool — I don't know how I've never heard of it before 🤯

**Comment by mcmire** [comment]
@guialbuk Yup you're totally right! Sometimes whenever you use Appraisal you see that sort of thing, but it looks in order for `shoulda-matchers` lines to appear in the individual gemfiles, we would have to have a `gemspec` line in the main Gemfile, which we do not. So, you're good!

**Comment by O330oei** [comment]
> > This looks good. Make sure to re-run `appraisal install` locally as the version shows up in all of the generated gemfiles as well. Other than that, feel free to release!
> 
> Hi @mcmire . I was running `appraisal install` but there are no `shoulda-matcher` versions in the generated Gemfiles, as unit tests directly require the gem code.
> 
> For acceptance tests, the gem is directly loaded from the filesystem:
> 
> https://github.com/thoughtbot/shoulda-matchers/blob/e0a0200fe47157c161fb206043540804bdad664e/spec/support/acceptance/adds_shoulda_matchers_to_project.rb#L32
> 
> So I think we don't need to update any generated gemfiles. Or maybe I'm missing something?

convert

**Pull request merged**